Communitter/src/main/java/com/communitter/api/controller/ImageController.java

```diff
@@ -0,0 +1,105 @@
+package com.communitter.api.controller;
+
+import com.communitter.api.model.Post;
+import com.communitter.api.model.User;
+import com.communitter.api.service.ImageService;
+import com.communitter.api.service.PostService;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.parameters.P;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import com.communitter.api.dto.ImageDTO;
+import com.communitter.api.dto.request.UserRequest;
+import com.communitter.api.service.ImageService;
+
+
+import java.io.IOException;
+
+@RestController
+@CrossOrigin
+@RequestMapping("/Images")
+public class ImageController {
+
+    private final ImageService imageService;
+
+    @Autowired
+    public ImageController(ImageService imageService) {
+        this.imageService = imageService;
+    }
+
+    // POST endpoint for uploading a user's profile picture
+    @PreAuthorize("@authorizer.authorizerForUser(#root,#userId)")
+    @PostMapping("/user/{userId}/profile-picture")
+    public ResponseEntity<String> uploadUserProfilePicture(@P ("userId") @PathVariable Long userId, @RequestParam("image") MultipartFile file) throws IOException {
+        String response = imageService.uploadProfilePicture(userId, file);
+        return ResponseEntity.ok(response);
+    }
+
+    
+    @GetMapping("/user/{userId}/profile-picture")
+    public ResponseEntity<?> downloadUserProfilePicture(@PathVariable Long userId) {
+    try {
```

Comment:
we should not need extra try catch blocks.

Reply:
revised thanks